hack/e2e.sh

```diff
@@ -0,0 +1,234 @@
+  ginkgo_args+=("--silence-skips")
+fi
+
+if [[ "${GINKGO_FORCE_NEWLINES}" == "y" ]]; then
+  ginkgo_args+=("--force-newlines")
+fi
+
+if [[ "${GINKGO_NO_COLOR}" == "y" ]]; then
+  ginkgo_args+=("--no-color")
+fi
+
+# The --host setting is used only when providing --auth_config
+# If --kubeconfig is used, the host to use is retrieved from the .kubeconfig
+# file and the one provided with --host is ignored.
+# Add path for things like running kubectl binary.
+PATH=$(dirname "${e2e_test}"):"${PATH}"
+export PATH
+
+# Choose the program to execute and additional arguments for it.
+#
+# Note that the e2e.test binary must have been built with "make DBG=1"
+# when using debuggers, otherwise debug information gets stripped.
+case "${E2E_TEST_DEBUG_TOOL:-ginkgo}" in
+  ginkgo)
+    program=("${ginkgo}")
+    if [[ -n "${GINKGO_PARALLEL_NODES:-}" ]]; then
+      program+=("--nodes=${GINKGO_PARALLEL_NODES}")
+    elif [[ ${GINKGO_PARALLEL} =~ ^[yY]$ ]]; then
+      program+=("--nodes=25")
+    fi
+    program+=("${ginkgo_args[@]:+${ginkgo_args[@]}}")
+    ;;
+  delve) program=("dlv" "exec") ;;
+  gdb) program=("gdb") ;;
+  *) kube::log::error_exit "Unsupported E2E_TEST_DEBUG_TOOL=${E2E_TEST_DEBUG_TOOL}" ;;
+esac
+
+# Move Ginkgo arguments that are understood by the suite when not using
+# the CLI.
+suite_args=()
+if [ "${E2E_TEST_DEBUG_TOOL:-ginkgo}" != "ginkgo" ]; then
+  for arg in "${ginkgo_args[@]}"; do
+    suite_args+=("--ginkgo.${arg#--}")
+  done
+fi
+
+# Generate full dumps of the test result and progress in <report-dir>/ginkgo/,
+# using the Ginkgo-specific JSON format and JUnit XML. Ignored if --report-dir
+# is not used.
+suite_args+=(--report-complete-ginkgo --report-complete-junit)
+
+# When SIGTERM doesn't reach the E2E test suite binaries, ginkgo will exit
+# without collecting information from about the currently running and
+# potentially stuck tests. This seems to happen when Prow shuts down a test
+# job because of a timeout.
+#
+# It's useful to print one final progress report in that case,
+# so GINKGO_PROGRESS_REPORT_ON_SIGTERM (enabled by default when CI=true)
+# catches SIGTERM and forwards it to all processes spawned by ginkgo.
+#
+# Manual invocations can trigger a similar report with `killall -USR1 e2e.test`
+# without having to kill the test run.
+GINKGO_CLI_PID=
+signal_handler() {
+  if [ -n "${GINKGO_CLI_PID}" ]; then
+    cat <<EOF
+
+*** $0: received $1 signal -> asking Ginkgo to stop.
+***
+*** Beware that a timeout may have been caused by some earlier test,
+*** not necessarily the one which gets interrupted now.
+*** See the "Spec runtime" for information about how long the
+*** interrupted test was running.
+
+EOF
+    # This goes to the process group, which is important because we
+    # need to reach the e2e.test processes forked by the Ginkgo CLI.
+    kill -TERM "-${GINKGO_CLI_PID}" || true
+
+    echo "Waiting for Ginkgo with pid ${GINKGO_CLI_PID}..."
+    wait "{$GINKGO_CLI_PID}"
```

Comment:
```suggestion
    wait "${GINKGO_CLI_PID}"
```

That's a mistake that I made in the original script.